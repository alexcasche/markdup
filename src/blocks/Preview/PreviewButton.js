import { Button } from 'elements'
import { tint, shade } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'

const MODIFIERS = {
  mobile: ({ theme }) => `
    color: pink;
    @media only screen and (min-width: ${theme.breakPoints.mobile}) {
      display: none;
    }
  `,
}

const PreviewButton = Button.extend`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
  margin-left: 10px;
  background-color: ${props => tint(0.25, props.theme.colors.grey)};
  border: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  border-radius: 2px;
  z-index: 5;
  &:hover {
    background-color: ${props => tint(0.35, props.theme.colors.grey)};
  }
  svg {
    color: ${props => shade(0.9, props.theme.colors.grey)};
    height: 14px;
    width: auto !important;
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default PreviewButton
