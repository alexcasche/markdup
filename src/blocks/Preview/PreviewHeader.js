import styled from 'styled-components'
import { tint } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'

const MODIFIERS = {
  expanded: () => `
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    padding: 0 10vw;
    z-index: 1;
  `,
}

const PreviewHeader = styled.header`
  display: flex;
  align-items: center;
  width: 100%;
  padding: 0 15px;
  background-color: #f7f7f7;
  border-top: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  border-bottom: 1px solid ${props => tint(0.5, props.theme.colors.grey)};
  overflow: visible;
  min-height: 46px;
  ${applyStyleModifiers(MODIFIERS)};
`

export default PreviewHeader
