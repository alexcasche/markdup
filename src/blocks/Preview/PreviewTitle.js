import styled from 'styled-components'
import { shade } from 'polished'

const PreviewTitle = styled.p`
  margin: 0;
  color: ${props => shade(0.9, props.theme.colors.grey)};
  font-size: 12px;
  font-weight: 500;
  text-transform: uppercase;
  flex-grow: 1;
`

export default PreviewTitle
