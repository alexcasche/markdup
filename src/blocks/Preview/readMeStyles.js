export default `
  h1 {
    padding-bottom: 0.3em;
    font-size: 32px;
    border-bottom: 1px solid #eaecef;
    font-weight: 500;
    line-height: 1.25;
    margin: 0.67em 0;
  }
  h2 {
    padding-bottom: 0.3em;
    font-size: 1.5em;
    border-bottom: 1px solid #eaecef;
    margin-top: 24px;
    margin-bottom: 16px;
    font-weight: 500;
    line-height: 1.25;
  }
  p {
    margin-top: 0;
    margin-bottom: 16px;
    line-height: 1.5;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Helvetica,
      Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji',
      'Segoe UI Symbol';
  }
  a {
    color: #0366d6;
    text-decoration: none;
  }
  a:hover {
    text-decoration: underline;
  }
  a:hover,
  a:active {
    outline-width: 0;
  }
  ul {
    padding-left: 2em;
    margin-top: 0;
    margin-bottom: 16px;
  }
  ul li + li {
    margin-top: 0.25em;
  }
  pre {
    padding: 16px;
    overflow: auto;
    line-height: 1.45;
    background-color: #f6f8fa;
    border-radius: 3px;
    font-size: 16px;
  }
  code {
    padding: 0.2em 0.4em;
    margin: 0;
    font-size: 85%;
    font-family: 'SFMono-Regular', Consolas, 'Liberation Mono', Menlo, Courier,
      monospace;
    background-color: #f6f8fa;
    border-radius: 3px;
  }
  table {
    display: block;
    width: 100%;
    overflow: auto;
    margin-top: 0;
    margin-bottom: 16px;
    border-spacing: 0;
    border-collapse: collapse;
  }
  tr {
    background-color: #fff;
    border-top: 1px solid #c6cbd1;
  }
  tr:nth-child(2n) {
    background-color: #f6f8fa;
  }
  th {
    text-align: center;
    font-weight: 600;
  }
  th,
  td {
    padding: 6px 13px;
    border: 1px solid #dfe2e5;
  }
  table code {
    padding: 0.2em 0.4em;
    margin: 0;
    font-size: 85%;
    background-color: rgba(27, 31, 35, 0.05);
    border-radius: 3px;
  }
`
