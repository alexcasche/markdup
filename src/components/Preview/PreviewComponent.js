import React from 'react'
import Highlight from 'react-highlight'
import { Preview } from 'blocks'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faExpandArrowsAlt,
  faEyeSlash,
  faTimes,
} from '@fortawesome/free-solid-svg-icons'
import { markdownToHtml } from 'utils/convertCodes'
import { PropTypes as t } from 'prop-types'

const PreviewComponent = props => {
  const { activeEditor, expanded, html, markdown, events } = props
  const preview = activeEditor === 'html' ? html : markdownToHtml(markdown)
  const expandedModifiers = [...(expanded ? ['expanded'] : [])]
  const expandIcon =
    expandedModifiers.indexOf('expanded') === -1 ? faExpandArrowsAlt : faTimes
  return (
    <Preview className="Preview" modifiers={expandedModifiers}>
      <Preview.Header modifiers={expandedModifiers}>
        <Preview.Title>Preview</Preview.Title>
        {!expanded ? (
          <Preview.Button onClick={events.togglePreview} modifiers={['mobile']}>
            <FontAwesomeIcon icon={faEyeSlash} />
          </Preview.Button>
        ) : null}
        <Preview.Button onClick={() => events.toggleExpand('preview')}>
          <FontAwesomeIcon icon={expandIcon} />
        </Preview.Button>
      </Preview.Header>
      <Preview.Content modifiers={expandedModifiers}>
        <Highlight innerHTML={true}>{preview}</Highlight>
      </Preview.Content>
    </Preview>
  )
}

PreviewComponent.propTypes = {
  activeEditor: t.string.isRequired,
  expanded: t.bool.isRequired,
  html: t.string.isRequired,
  markdown: t.string.isRequired,
  events: t.shape({
    toggleExpand: t.func.isRequired,
    togglePreview: t.func.isRequired,
  }).isRequired,
}

export default PreviewComponent
