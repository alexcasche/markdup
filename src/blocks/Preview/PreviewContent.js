import styled from 'styled-components'
import { tint } from 'polished'
import { applyStyleModifiers } from 'styled-components-modifiers'
import readMeStyles from './readMeStyles'

const MODIFIERS = {
  expanded: () => `
    height: calc(100% - 46px);
    margin-top: 46px;
    padding-top: 34px;
    padding-right: calc(10vw - 15px);
    padding-bottom: 60px;
    padding-left: 10vw;
  `,
}

const PreviewContent = styled.div`
  height: calc(100% - 46px);
  padding: 15px;
  overflow-y: scroll;
  word-wrap: break-word;
  color: ${props => props.theme.colors.black};
  font-family: ${props => props.theme.text.fontCode};
  font-size: 16px;
  letter-spacing: normal;
  line-height: 1.5;
  user-select: text;
  &::-webkit-scrollbar {
    width: 15px;
    position: absolute;
  }
  &::-webkit-scrollbar-track {
    background-color: #f7f7f7;
  }
  &::-webkit-scrollbar-thumb {
    background-color: ${props => tint(0.35, props.theme.colors.grey)};
  }
  &::-webkit-scrollbar-thumb:hover {
    background-color: ${props => tint(0.5, props.theme.colors.grey)};
  }
  & > div > *:first-child {
    margin-top: 0;
  }
  ${applyStyleModifiers(MODIFIERS)};
  ${readMeStyles};
`

export default PreviewContent
