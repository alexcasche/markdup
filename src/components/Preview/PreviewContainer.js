import React from 'react'
import { connect } from 'react-redux'
import { getCurrentWorkspaceVal, getSettingVal } from 'selectors'
import PreviewComponent from './PreviewComponent'

export const PreviewContainer = props => {
  return <PreviewComponent {...props} />
}

function mapStateToProps(state) {
  return {
    activeEditor: getSettingVal(state, 'activeEditor'),
    html: getCurrentWorkspaceVal(state, 'html'),
    markdown: getCurrentWorkspaceVal(state, 'markdown'),
  }
}

export default connect(mapStateToProps)(PreviewContainer)
