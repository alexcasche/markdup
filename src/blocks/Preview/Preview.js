import styled from 'styled-components'

import PreviewButton from './PreviewButton'
import PreviewContent from './PreviewContent'
import PreviewHeader from './PreviewHeader'
import PreviewTitle from './PreviewTitle'

const Preview = styled.div`
  height: 100%;
  background-color: white;
  overflow: hidden;
`

Preview.Button = PreviewButton
Preview.Content = PreviewContent
Preview.Header = PreviewHeader
Preview.Title = PreviewTitle

export default Preview
