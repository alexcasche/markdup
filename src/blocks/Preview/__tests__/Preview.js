import Preview from '../Preview'

describe('Preview Element', () => {
  it('renders correctly', () => {
    const tree = render(<Preview />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
