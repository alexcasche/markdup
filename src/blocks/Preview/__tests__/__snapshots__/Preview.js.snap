// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Element renders correctly 1`] = `
.c0 {
  height: 100%;
  background-color: white;
  overflow: hidden;
}

<div
  className="c0"
/>
`;
